src/main/java/org/spongepowered/api/event/action/CreateMapEvent.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.event.action;
+
+import org.spongepowered.api.event.Cancellable;
+import org.spongepowered.api.event.Event;
+import org.spongepowered.api.map.MapInfo;
+
+/**
+ * Called when a new map is registered.
+ * Can be caused by players or a plugin.
+ */
+public interface CreateMapEvent extends Event, Cancellable {
+    /**
+     * Gets the {@link org.spongepowered.api.map.MapInfo} for this map
+     * @return MapInfoData Map data
+     */
+    MapInfo getMapInfo();
```

Comment:
Is it possible to change the map info instance? Or are we to expect it being mutable from the get go?

Reply:
MapInfo is mutable, but it can refer to more than one map. I can see that maybe you would want to swap out the event's MapInfo with your own that already refers to a lot of maps. I think the thing that would make the most sense to cancel the creation of the new map, then get the used item (assuming there was one), and switch it out for a ItemStack with type filled_map, and then offer it your map info, because the event is for the creation of a new map, not the item use.